posthog/api/dashboards/dashboard.py

```diff
@@ -419,14 +422,26 @@ def get_tiles(self, dashboard: Dashboard) -> Optional[list[ReturnDict]]:
         )
         self.user_permissions.set_preloaded_dashboard_tiles(list(tiles))
 
-        for tile in tiles:
-            self.context.update({"dashboard_tile": tile})
+        # Sort tiles by layout to ensure insights are computed in order of appearance on dashboard
+        sorted_tiles = sorted(
+            tiles,
+            key=lambda tile: (
+                tile.layouts.get("xs", {}).get("y", 0),
+                tile.layouts.get("xs", {}).get("x", 0),
+            ),
+        )
+
+        # In case of a large number of tiles on a dashboard,
+        # ensure all tiles are computed one at a time to avoid overwhelming the database
```

Comment:
I'm confused, i don't think this code actually runs any queries?

Reply:
Dashboard refresh results in this call to `DashboardTileSerializer` goes to `InsightSerializer` in which the [get_result](https://github.com/PostHog/posthog/blob/master/posthog/api/insight.py#L560) function is triggered to compute the insight. In case result is cached, we get the results in the dashboard refresh request, otherwise we just get the query id to poll for to get the insight results.

Using `task_chain_context` before the calls to `DashboardTileSerializer` results in all the insight queries getting chained,

Reply:
ah but in that case this will just fire off all the queries at the same time anyway right?

Reply:
Yes but this time all the queries will be in a celery chain. So if one user refreshes the dashboard, they'll at max only be able to use up one clickhouse thread. Previously they would have used up a lot more threads ([limited by the concurrency on the celery task](https://github.com/PostHog/posthog/blob/feat/dash-apply-preview/posthog/tasks/tasks.py#L57)/queue). Noticed it just got bumped up to 50 😅